src/Assets.jl

```diff
@@ -14,22 +30,27 @@ Cash(S::Symbol) = Cash{S,unit(S)}()
 Cash(::Currency{S}) where {S} = Cash(S)
 
 unit(::Cash{C,N}) where {C,N} = N
-code(c::Cash) = code(currency(c))
-name(c::Cash) = name(currency(c))
+currency(::Cash{C}) where {C} = C
+code(c::Cash)   = code(currency(c))
+name(c::Cash)   = name(currency(c))
+symbol(c::Cash) = symbol(currency(c))
 
 Base.show(io::IO, ::Cash{C}) where {C} = print(io, string(C))
 
 function Position{Cash{C,N}}(a) where {C,N}
     T = FixedDecimal{Int,N}
     Position{Cash{C,N},T}(T(a))
 end
+Position(c::C, v) where {C<:Cash} = Position{typeof(c)}(v)
 
 """
 `ListedEquity` is a financial instrument represented by a singleton type with the exchange it is listed on, and its symbol (such as :MSFT)
 """
-struct ListedEquity{E,S,C} <: Instrument{S,Currency{C}}
-    ListedEquity(E::Symbol, S::Symbol, C::Symbol) = new{E,S,C}()
+struct ListedEquity{E,S} <: Instrument{S,currency(E)}
```

Comment:
We might consider renaming this from `ListedEquity` to `Stock` or more precisely `CommonStock`. "Listed equity" encompasses things like ETFs that are not common stocks.

Reply:
OK, it's like 4 or 5 tests, but I'll change them.